crates/miden-lib/asm/miden/auth/multisig.masm

```diff
@@ -0,0 +1,284 @@
+
+        sub.1 dup push.0.0.0 loc_load.PUB_KEY_MAP_IDX
+        # => [owner_key_slot, [0, 0, 0, i-1], i-1, MSG]
+
+        exec.account::get_map_item
+        # => [OWNER_PUB_KEY, i-1, MSG]
+
+        loc_storew.CURRENT_PK
+        # => [OWNER_PUB_KEY, i-1, MSG]
+
+        # Check if signature exists for this signer.
+        # -----------------------------------------------------------------------------------------
+
+        movup.4 movdn.8 swapw dupw movdnw.2
+        # => [MSG, OWNER_PUB_KEY, MSG, i-1]
+
+        hmerge
+        # => [SIG_KEY, MSG, i-1]
+
+        adv.has_mapkey
+        # => [SIG_KEY, MSG, i-1]
+
+        adv_push.1
+        # => [has_signature, SIG_KEY, MSG, i-1]
+
+        # if SIG_KEY => SIGNATURE exists in AdviceMap check the signature
+        if.true
+            # => [SIG_KEY, MSG, i-1]
+
+            dupw.1
+            # => [MSG, SIG_KEY, MSG, i-1]
+
+            swapw
+            # => [SIG_KEY, MSG, MSG, i-1]
+
+            # Verify the signature against the public key and message.
+            # -----------------------------------------------------------------------------------------
+
+            loc_loadw.CURRENT_PK
+            # => [PK, MSG, MSG, i-1]
+
+            swapw
+            # => [MSG, PK, MSG, i-1]
+
+            # Emit the authentication request event that pushes a signature for the message to the advice stack.
+            emit.AUTH_REQUEST
+
+            swapw
+            # OS => [PUB_KEY, MSG, MSG, i-1]
+            # AS => [SIGNATURE]
+
+            exec.rpo_falcon512::verify
+            # => [MSG, i-1]
+
+            loc_load.SUCCESSFUL_VERIFICATIONS
+            add.1
+            loc_store.SUCCESSFUL_VERIFICATIONS
+            # => [MSG, i-1]
+        else
+            dropw
+            # => [MSG, i-1]
+        end
+        # => [MSG, i-1]
+
+        movup.4
+        # => [i-1, MSG]
+
+        dup neq.0
+        # => [should_continue, i-1, MSG]
+    end
+    # => [i-1, MSG]
+
+    # Check number of successful signature verifications
+    # -----------------------------------------------------------------------------------------
+
+    drop
+    # => [MSG]
+
+    loc_load.SUCCESSFUL_VERIFICATIONS
+    # => [num_verified_signatures, MSG]
+end
+
+#! Authenticate a transaction using the Falcon signature scheme with multi-signature support.
+#!
+#! This procedure implements multi-signature authentication by:
+#! 1. Computing the transaction summary message that needs to be signed
+#! 2. Verifying signatures from multiple required signers against their public keys
+#! 3. Ensuring the minimum threshold of valid signatures is met
+#! 4. Implementing replay protection by tracking executed transactions
+#!
+#! The procedure first increments the account nonce, then computes and verifies signatures for
+#! the following message (in memory order):
+#! [ACCOUNT_DELTA_COMMITMENT, INPUT_NOTES_COMMITMENT,
+#!  OUTPUT_NOTES_COMMITMENT, [0, 0, ref_block_num, final_nonce]]
```

Comment:
This is not how we build the `SALT` in the procedure. We can probably delete this as we correctly explain how `SALT` works on lines 217 - 219 below.